airflow/contrib/hooks/winrm_hook.py

```diff
@@ -17,121 +17,191 @@
-    :type key_file: str
-    :param timeout: timeout for the attempt to connect to the remote_host.
-    :type timeout: int
-    :param keepalive_interval: send a keepalive packet to remote host
-        every keepalive_interval seconds
-    :type keepalive_interval: int
+    :param str remote_host: Remote host to connect to.
+        Ignored if `endpoint` is not `None`.
+    :param int remote_port: Remote port to connect to.
+        Ignored if `endpoint` is not `None`.
+    :param str endpoint: When set to `None`, endpoint will be constructed like this:
+        'http://{remote_host}:{remote_port}/wsman'
+    :param str username: username to connect to the remote_host
+    :param str password: password of the username to connect to the remote_host
     """
-
     def __init__(self,
                  ssh_conn_id=None,
+                 endpoint=None,
                  remote_host=None,
+                 remote_port=5985,
+                 transport='plaintext',
                  username=None,
                  password=None,
-                 key_file=None,
-                 timeout=10,
-                 keepalive_interval=30
+                 service='HTTP',
+                 keytab=None,
+                 ca_trust_path=None,
+                 cert_pem=None,
+                 cert_key_pem=None,
+                 server_cert_validation='validate',
+                 kerberos_delegation=False,
+                 read_timeout_sec=30,
+                 operation_timeout_sec=20,
+                 kerberos_hostname_override=None,
+                 message_encryption='auto',
+                 credssp_disable_tlsv1_2=False,
+                 send_cbt=True,
                  ):
         super(WinRMHook, self).__init__(ssh_conn_id)
-        # TODO make new win rm connection class
         self.ssh_conn_id = ssh_conn_id
+        self.endpoint = endpoint
         self.remote_host = remote_host
+        self.remote_port = remote_port
+        self.transport = transport
         self.username = username
         self.password = password
-        self.key_file = key_file
-        self.timeout = timeout
-        self.keepalive_interval = keepalive_interval
-        # Default values, overridable from Connection
-        self.compress = True
-        self.no_host_key_check = True
+        self.service = service
+        self.keytab = keytab
+        self.ca_trust_path = ca_trust_path
+        self.cert_pem = cert_pem
+        self.cert_key_pem = cert_key_pem
+        self.server_cert_validation = server_cert_validation
+        self.kerberos_delegation = kerberos_delegation
+        self.read_timeout_sec = read_timeout_sec
+        self.operation_timeout_sec = operation_timeout_sec
+        self.kerberos_hostname_override = kerberos_hostname_override
+        self.message_encryption = message_encryption
+        self.credssp_disable_tlsv1_2 = credssp_disable_tlsv1_2
+        self.send_cbt = send_cbt
```

Comment:
Please add docstrings for all the new parameters you add.

Reply:
I have a 'seealso' that points to the pywinrm documentation that describes the parameters in detail since I'm just passing them through. Do you want me to reproduce it here too?

Reply:
I get your point but it would be ideal to see what those parameters are in the airflow docs itself than to open an external code file to check it.

Reply:
I copied the descriptions from pywinrm into the Airflow hook.